docs/diga_api.md

```diff
@@ -0,0 +1,289 @@
+which also contains a section about requesting certificates from ITSG trust center.
+
+## Verfahren
+
+The `verfahren` parameter consists of 5 characters describing the kind of the request.
+The first character is either `T` or `E` denoting `Testdaten` (test data) or `Echtdaten` (real data) - [attachment 4 - Verfahrenskennungen](https://www.gkv-datenaustausch.de/media/dokumente/standards_und_normen/technische_spezifikationen/Anlage_4_-_Verfahrenskennungen.pdf) on [gkv technische standards page](https://www.gkv-datenaustausch.de/technische_standards_1/technische_standards.jsp).
+
+The next 3 characters are used to discriminate between:
+
+- verification - DFC (**D**iga **F**reischalt**c**ode)
+- billing - DRE (**D**iga **Re**chnung).
+
+The last one is for versioning which leaves us with the following values:
+
+- `EDFC0` and `TDFC0` for verification
+- `EDRE0` and `TDRE0` for billing.
+
+## Nutzdaten
+
+`nutzdaten` is the tricky field which contains verification or billing related information based on the `verfahren` value.
+`nutzdaten` are defined in an xml sheet which needs to be encrypted and encoded as byte array before sending the actual requests.
+The structure of the xml files are defined by `xsd` files which can be found in `src/main/resources` or downloaded from [gkv page](https://www.gkv-datenaustausch.de/leistungserbringer/digitale_gesundheitsanwendungen/digitale_gesundheitsanwendungen.jsp) as [attachment - DiGA-Abrechnungsschema (XSD) (ZIP)](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/digitale_gesundheitsanwendungen/technische_anlagen_aktuell_7/DiGA_XSD_20201029.zip).
+Examples for the data and additional details on the encryption are described in the following.
+
+# Request details
+
+## Code verification requests
+
+See `EDFC0-Pruefung_2.0.0.xsd` file for reference:
+
+Data for sending the request:
+
+```xml
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<Pruefung_Freischaltcode xmlns="http://www.gkv-datenaustausch.de/XML-Schema/EDFC0_Pruefung/2.0.0"
+                         version="002.000.000" gueltigab="2020-07-01" verfahrenskennung="TDFC0" nachrichtentyp="ANF"
+                         absender="987654321" empfaenger="123456789">
+    <Antwort>
+        <IK_DiGA_Hersteller>123456789</IK_DiGA_Hersteller>
+        <IK_Krankenkasse>987654321</IK_Krankenkasse>
+        <DiGAID>12345000</DiGAVEID>
+        <Freischaltcode>ABCDEFGHIJKLMNOP</Freischaltcode>
+    </Antwort>
+</Pruefung_Freischaltcode>
+```
+
+In case you are not an approved DiGA manufacturer yet, you can put `12345` as `DiGAID`.
+You should already have an `IK` number though as otherwise, you also wouldn't have the certificate for encryption/decryption.
```

Comment:
Or is there a way to make test requests without IK number and certificate?

Reply:
This is what I expected :-/ It makes it quite tough to get started with the DiGA api and this library.

Therefore, it could make sense to outline this in the beginning that these two things are blockers and should be requested early. Could even make sense in the README under prerequisites as first point?

```
Sending requests to the DiGA api requires companies to request an IK number and certificates for encryption/decryption from ITSG beforehand. These things cannot be requested in parallel and it take up to three weeks to get them. Therefore, it makes sense to request them as early as possible. Please see the {wiki | docs folder} for more information.
```

Reply:
Updated the `Get Started` section in the readme.